parsers/wiki_parser.py

```diff
@@ -0,0 +1,291 @@
+    Pages for heroes are in the form of `Hero name/Responses`. We need only the `Hero name` part for heroes.
+
+    :param hero_page: hero's responses page as string.
+    :return: Hero name as parsed
+    """
+    return hero_page.split('/')[0]
+
+
+def create_responses_text_and_link_list(responses_source):
+    """Method that for a given source of a hero's response page creates a list of tuple: (original_text, 
+    processed_text, link).
+    Steps involved:
+    * Use regex to find all lines containing mp3 files and responses.
+    * Process it to get original response text and file name.
+    * Create a list of files and get all the links for them by calling `links_for_files`.
+    * Process original text to get processed response.
+    * Add original response text, processed response text and file link to a list as a tuple.
+
+    :param responses_source: Mediawiki source
+    :return: list with tuples of (original_text, processed_text, link).
+    """
+    responses_list = []
+    file_and_text_list = []
+
+    response_regex = re.compile(RESPONSE_REGEX)
+    file_regex = re.compile(FILE_REGEX)
+
+    for response in response_regex.finditer(responses_source):
+        original_text = parse_response(response['text'])
+        if original_text == '':
+            pass
+
+        files_source = response['files']
+        for file in file_regex.finditer(files_source):
+            file_name = file['file'].replace('_', ' ').capitalize()
+            file_and_text_list.append([original_text, file_name])
+
+    files_list = [file for text, file in file_and_text_list]
+    file_and_link_dict = links_for_files(files_list)
+
+    for original_text, file in file_and_text_list:
+        processed_text = clean_response_text(original_text)
+        if processed_text != '':
+            try:
+                link = file_and_link_dict[file]
+                responses_list.append((original_text, processed_text, link))
+            except KeyError:
+                # Ignore files with no links to mp3 files. Happens to broken files and files undergoing migration.
+                pass
+
+    return responses_list
+
+
+def parse_response(text):
+    # Special cases
+    if '(broken file)' in text:
+        return ''
+    if 'versus (TI ' in text:
+        return ''
+    if 'Ceeeb' in text:
+        return ''
+
+    text = re.sub(r'…', '...', text)  # Replace ellipsis with three dots
+
+    regexps_empty_sub = [r'<!--.*?-->',  # Remove comments
+                         r'{{resp\|(r|u|\d+|d\|\d+)}}',  # Remove response rarity
+                         r'{{hero icon\|[a-z- \']+\|\d+px}}',  # Remove hero icon
+                         r'{{item( icon)?\|[a-z0-9() \']+\|\d+px}}',  # Remove item icon
+                         r'\[\[File:[a-z.,!\'() ]+\|\d+px\|link=[a-z,!\'() ]+]]',  # Remove Files
+                         r'<small>\[\[#[a-z_\-\' ]+\|\'\'followup\'\']]</small>',
+                         # Remove followup links in <small> tags
```

Comment:
This comment should be above the line it is related to, if it cannot fit in the same line.

Reply:
This was due to IDE auto formatting as the line exceeded 120 char limit. Fixed in d266d930fbd4885d99aa0bd34300009ef2e3340c